src/tracing/tracing.c

```diff
@@ -37,11 +38,19 @@ static xdebug_trace_handler_t *xdebug_select_trace_handler(int options)
 		case 0: tmp = &xdebug_trace_handler_textual; break;
 		case 1: tmp = &xdebug_trace_handler_computerized; break;
 		case 2: tmp = &xdebug_trace_handler_html; break;
+		case 3: tmp = &xdebug_trace_handler_flamegraph_cost; break;
+		case 4: tmp = &xdebug_trace_handler_flamegraph_mem; break;
 		default:
 			php_error(E_NOTICE, "A wrong value for xdebug.trace_format was selected (%d), defaulting to the textual format", (int) XINI_TRACE(trace_format));
 			tmp = &xdebug_trace_handler_textual; break;
 	}
 
+	if (options & XDEBUG_TRACE_OPTION_FLAMEGRAPH_COST) {
```

Comment:
This is needed so that people can pass in the requested format with [xdebug_start_function_trace()](https://xdebug.org/docs/trace#xdebug_start_trace). The switch should take care of selecting the right handler from the `xdebug.trace_format` INI setting, but this is needed because the function doesn't take the format as a number, but just as a bit field.

It also seems that `*_HTML` is missing from here, but I don't really mind that (it's a stupid format).

Reply:
Ok I'll dix that then.

Reply:
> but I don't really mind that (it's a stupid format).

Isn't it the default ? If not, I mean, stupid format or not this piece of code should remain consistent in order to avoid confusion for users (if anyone uses this).

Reply:
`html` isn't missing, only the `textual` format is, but it is set in the `default` case, so it is OK.